sim/sim.go

```diff
@@ -94,6 +94,17 @@ func (s *Simulation) Run(instanceCount uint64, maxRounds uint64) error {
 			// The last incomplete instance is used for testing assertions.
 			currentInstance = s.ec.BeginInstance(*decidedChain, pt)
 
+			// Simulate the behaviour of an honest integrator, where upon receiving finality
+			// certificates from future rounds it signals gpbft to skip ahead.
```

Comment:
This will react much faster than we might expect in reality, but it's probably good enough for exercising the code.

Reply:
Right 👍 

I have left a TODO in the inner clause to later enhance the simulation options once the finality certificate propagation protocol/mechanism is clearer. Happy to add delay now if you think this is too unrealistic.